UKHO.D365CallbackDistributorStub.API/UKHO.D365CallbackDistributorStub.API/Services/CallbackService.cs

```diff
@@ -42,5 +54,56 @@ public static bool SaveCallbackRequest(CallbackRequest callbackRequest, string s
             }
 
         }
+
+        public bool SaveCommandCallbackRequest(string subscriptionId, HttpStatusCode? httpStatusCode)
+        {
+            try
+            {
+                CommandCallbackRequest? commandCallbackRequest = s_commandCallbackRequestList.Where(a => a.SubscriptionId == subscriptionId).LastOrDefault();
+
+                if (commandCallbackRequest != null)
+                {
+                    if (httpStatusCode == null)
+                    {
+                        s_commandCallbackRequestList.Remove(commandCallbackRequest);
+                    }
+                    else
+                    {
+                        commandCallbackRequest.HttpStatusCode = httpStatusCode ?? NoContent;
+                    }
+                }
+                else
+                {
+                    if (httpStatusCode != null)
+                    {
+                        s_commandCallbackRequestList.Add(new CommandCallbackRequest
+                        {
+                            SubscriptionId = subscriptionId,
+                            HttpStatusCode = httpStatusCode ?? NoContent,
+                        });
+                    }
+                    else
+                    {
+                        _logger.LogInformation("Request not found in memory for subscriptionId: {subscriptionId}", subscriptionId);
+                        return true;
+                    }
+                }
+
+                if (s_commandCallbackRequestList.Count >= 50)
+                {
+                    s_commandCallbackRequestList.RemoveAt(0);
+                }
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public CommandCallbackRequest? SubscriptionInCommandCallbackList(string subscriptionId)
+        {
+           return s_commandCallbackRequestList.Where(a => a.SubscriptionId == subscriptionId).LastOrDefault();
```

Comment:
Please simplify LINQ expression as below:
```suggestion
           return s_commandCallbackRequestList.LastOrDefault(a => a.SubscriptionId == subscriptionId);
```

Reply:
Fixed